### ISSUE TITLE
akka-http-jackson and akka-serialization-jackson shall share the same jackson-json ObjectMapper.

### ISSUE THREAD

**Issue by yunglin** [akka/akka-http#3168]
https://github.com/akka/akka-http/blob/21a9760920dffda1f3425377b282df86fe5613da/akka-http-marshallers-java/akka-http-jackson/src/main/java/akka/http/javadsl/marshallers/jackson/Jackson.java#L21

Jackson-json's ObjectMapper is high configurable. For the same object, object mappers with different configuration can produce totally different json messages. [akka-serialization-jackson](https://github.com/akka/akka/blob/master/akka-serialization-jackson/src/main/resources/reference.conf#L39) ship with a high customized object mapper. For this reason, akka-http-jackson shall reuse this object mapper when possible. So that akka-http and akka-remote can generate the consistent json message for the same object.

**Comment by raboof** [comment]
I agree it should be possible to use the same `ObjectMapper` for Akka Remoting serialization and the Akka HTTP API serialization.

However, I think by default they should be separate: changing the Remoting serialization should not change your web API if you didn't explicitly opt-in to that.

**Comment by yunglin** [comment]
How about making the objectMapper in akka-http a config option and allow users to point this config to akka-serialization-jackson's object mapper?

**Comment by raboof** [comment]
> How about making the objectMapper in akka-http a config option and allow users to point this config to akka-serialization-jackson's object mapper?

I would not be opposed to this, but I suspect it will be hard to implement (elegantly) because Jackson marshalling is currently provided as static methods on akka.http.javadsl.marshallers.jackson.Jackson, which we can't easily inject the settings into.
